### DISCUSSION TITLE
SteamDeck Development Support

### DISCUSSION THREAD

**Discussion by spkane** [89luca89/distrobox#1236] [Ideas]
So, after a bit of effort, I was able to find a workflow that allows me to reasonably easily use `distrobox` to do some SteamDeck development (on the SteamDeck) while leaving the root filesystem read-only and having matching versions of libraries and what-not in the container.

After a lot of searching, I came across this Git repo [SteamDeckHomebrew/holo-docker](SteamDeckHomebrew/holo-docker), which does a very good job of creating a Linux container from the SteamDeck images.

For the moment, I have forked it and created a branch which has some changes to make the build process work completely on the SteamDeck , if desired, and adds some flexibility to define the image name and tag [spkane/steamdeck-holo-container](https://github.com/spkane/steamdeck-holo-container/tree/spkane). I have submitted a [PR to the upstream](https://github.com/SteamDeckHomebrew/holo-docker/pull/9), but I have no idea yet if it will be accepted.

However, you can use their image(`ghcr.io/steamdeckhomebrew/holo-base:latest`), my image(`docker.io/spkane/steamdeck-holo-base`), or build your own, and then something like this appears to work very well with `distrobox`.

```sh
distrobox create --image ghcr.io/steamdeckhomebrew/holo-base:latest -n steamos
distrobox enter steamos
```

And now, in general, you can install packages and build apps that should be capable of running natively on the SteamDeck outside the container.

**NOTE**: I did see a few of `pacman`'s **post-transaction hooks** failing when I was installing new packages, but in general, this appeared only to be when they were trying to change the permissions of files that were mounted into the container, and this did not appear to cause any issues.

**Comment by 89luca89** [comment]
You can use `ghcr.io/linuxserver/steamos:latest` to create a full deck

`distrobox create -n steamos --image ghcr.io/linuxserver/steamos:latest --init --unshare-all`

Then

`distrobox enter steamos -- INTEL_DEBUG=norbc gamescope -e -f -b -g -W 1280 -H 720 -r 60 -- /usr/bin/steam -steamdeck`

And this will launch a steam in deck mode in a window, granted you need to be on Wayland on the host so that gamescope can work

**Reply by spkane** [reply]
The problem with this from my testing is that the `linuxserver/steamos` container is built on top of the newest Arch Linux base image instead of being identical to what is shipping on the SteamDeck.

The approach mentioned above uses the recovery image for the Steam Deck to build the container so that the versions of critical libraries like `glibc` and such are actually in sync with what is being used on the SteamDeck. Library version mismatches can lead to compatibility issues if the compilation step isn't against the identical versions as the runtime will use.
